src/components/NewsletterSignup/NewsletterSignup.stories.tsx

```diff
@@ -0,0 +1,131 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { Box, VStack } from "@chakra-ui/react";
+import { withDesign } from "storybook-addon-designs";
+import NewsletterSignup from "./NewsletterSignup";
+import { sectionTypeArray } from "../../helpers/types";
+import Heading from "../Heading/Heading";
+
+const meta: Meta<typeof NewsletterSignup> = {
+  title: "Components/Form Elements/NewsletterSignup",
+  component: NewsletterSignup,
+  decorators: [withDesign],
+  parameters: {
+    jest: ["NewsletterSignup.test.tsx"],
+  },
+  argTypes: {
+    className: { control: false },
+    confirmationHeading: {
+      control: "text",
+    },
+    confirmationText: {
+      control: "text",
+    },
+    descriptionText: {
+      control: "text",
+      table: {
+        defaultValue: {
+          summary:
+            "Stay connected with the latest research news from NYPL, including information about our events, programs, " +
+            "exhibitions, and collections.",
+        },
+      },
+    },
+    formHelperText: {
+      control: "text",
+    },
+    id: { control: false },
+    newsletterSignupType: {
+      control: "select",
+      options: sectionTypeArray,
+      table: {
+        defaultValue: {
+          summary: "whatsOn",
+        },
+      },
+    },
+    onChange: { control: false },
+    onSubmit: { control: false },
+    title: {
+      control: "text",
+      table: { defaultValue: { summary: "Sign Up for Our Newsletter!" } },
+    },
+    valueEmail: { control: false },
+    view: {
+      control: "select",
+      options: ["form", "submitting", "confirmation", "error"],
+      table: {
+        defaultValue: {
+          summary: "form",
+        },
+      },
+    },
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof NewsletterSignup>;
+
+/**
+ * Main Story for the NewsletterSignup component. This must contains the `args`
+ * and `parameters` properties in this object.
+ */
+export const WithControls: Story = {
+  args: {
+    className: undefined,
+    confirmationHeading: "Thank you for signing up!",
+    confirmationText:
+      "You can update your email subscription preferences at any time using the links at the bottom of the email.",
+    descriptionText: undefined,
+    formHelperText: undefined,
+    id: undefined,
+    isInvalidEmail: false,
+    newsletterSignupType: undefined,
+    onChange: undefined,
+    onSubmit: undefined,
+    title: undefined,
+    valueEmail: undefined,
+    view: undefined,
+  },
+  parameters: {
+    design: {
+      type: "figma",
+      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?node-id=80849%3A174194&mode=dev",
+    },
+    jest: "NewsletterSignup.test.tsx",
+  },
+  render: (args) => <NewsletterSignup {...args} />,
+};
+
+export const BrowserStates: Story = {
```

Comment:
Can you include this in the `.mdx`  file?

Also, another story that explicitly displays the confirmation and error views would be helpful, _or_ include those in this story and maybe rename this to "Component states"?

Reply:
Added.